### DISCUSSION TITLE
blog/two-way-anova-in-r/

### DISCUSSION THREAD

**Discussion by giscus** [AntoineSoetewey/statsandr#106] [General]
# blog/two-way-anova-in-r/

Learn how to do a two-way ANOVA in R. You will also learn its aim, hypotheses, assumptions, and how to interpret the results of the two-way ANOVA

https://statsandr.com/blog/two-way-anova-in-r/

**Comment by technocrat** [comment]
Thanks for finally clearing that up. I had learned much of this piece wise but never connected the dots coherently.

**Reply by AntoineSoetewey** [reply]
Glad you find it useful @technocrat !

**Comment by 003136** [comment]
This was extremely helpful! My mentor asked that I rerun the tests, but control for two covariates (Age and Gender). If there a way to add covariates to the ggbetweenstats function for it runs an ANCOVA instead of an ANOVA?

**Reply by AntoineSoetewey** [reply]
Hello,

No unfortunately the `{ggstatsplot}` package does not allow to run ANCOVA nor two-way ANOVA.

These two posts may be helpful in your case:

- [Two-way ANOVA in R](https://statsandr.com/blog/two-way-anova-in-r/)
- [Multiple linear regression in R](https://statsandr.com/blog/multiple-linear-regression-made-simple/)


Hope this helps.

Regards,
Antoine

**Comment by kowalikmarcel** [comment]
This blog is marvellous. I'm so glad that I found you! Thank you for sharing the knowledge 😍

**Reply by AntoineSoetewey** [reply]
Thank you for your kind words @Marcyyyl !

**Comment by fiuzatayna** [comment]
Hey, thank you for the walkthrough! Much appreciated. :smile:

I was wondering if you know of any visualization package like `ggstatsplot` that accepts working with `Dunnett's test` results.

**Reply by AntoineSoetewey** [reply]
Dear @fiuzatayna,

It seems to be on the to do list of the maintainer of the {ggstatsplot} package (see https://github.com/IndrajeetPatil/ggstatsplot/issues/500).

There is also the {ggpubr} package, but as far as I know, I don't think there is a possibility to display results of Dunnett's test.

In the meantime, you could try adding the p-values manually to a ggplot with `annotate()` or `geom_text()`. You can also check this tutorial https://www.datanovia.com/en/blog/ggpubr-how-to-add-p-values-generated-elsewhere-to-a-ggplot/, but it is not as straightforward and easy as with {ggstatsplot}.

If you find a good way, feel free to let me know; I am also interested in this feature!

Regards,
Antoine

**Comment by AndrewSalvatore** [comment]
Hello,

I'm trying to understand the process of deciding upon and implementing type 1, 2, or 3 ANOVA. I explored the external link you provided but am still having difficulties understanding this concept.

In your example, you use a type 1 ANOVA, which assumes a balanced design. However, the sample sizes of sex and species in the penguin dataset are unbalanced. Could you elaborate on this?

Additionally, you say that type 2 should be used when there are no significant interactions and type 3 when there are. In that case, one would have to conduct an ANOVA first to know if a significant interaction exists or not. Does this mean you first conduct a type 1 ANOVA, and then reconduct a type 2 or 3 based on the results of the first ANOVA?

Any help would be greatly appreciated,
Andrew

**Reply by AntoineSoetewey** [reply]
Dear @AndrewSalvatore,

Thanks for your questions.

You are right: type I ANOVA assumes a balanced design, and this is the reason I wrote "we assume a balanced design here" (even though it is unbalanced, but I wanted to use the penguins dataset for this post).

Regarding your second question about the interaction: it's best to check whether there is an interaction, and then use the appropriate type of ANOVA. What I usually do is a type III ANVOVA (or a linear model with interaction (i.e., `lm(y ~ a * b)`) and keep it if the interaction is significant, otherwise simplify the model by running a type II ANOVA (without the interaction).

Hope this is more clear.

Regards,
Antoine

**Comment by nop9898** [comment]
Thank you for your extremely helpful blog.

**Reply by AntoineSoetewey** [reply]
You're welcome @nop9898!
